### DISCUSSION TITLE
Add ability for employees to attach files to order

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#35586] [Ideas]
#### Is your feature request related to a problem?

In real life, a lot of orders don't go as planned. Something is wrong, something is changing, customer wants something extra you don't have in stock, customer wants to modify something etc.

Until recently, there was no way to store any information regarding the order in Prestastashop. Order note is a HUGE HELP. What would make this perfect would be, if a employee can attach files to the order.

#### Describe the solution you'd like

Add ability for an employee to attach files to an order - PDFs, zips, images. Legal documents, confirmations, saved email communication, drawings etc.

#### Design proposal

![files](https://user-images.githubusercontent.com/6097524/108507615-e536f600-72ba-11eb-9740-8b457f20e263.JPG)

#### Technical solution

- Add new database table and entity - **OrderAttachment** with id_attachment, id_order, file, file_name, file_size, mime.
- Add a new section to order BO screen with a file upload input.
- On page load, fetch all the attachments regarding this order, show file name as a link and two buttons to delete a view the attachment.

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Well, that's a huge feature for the Order and it seems useful for the merchant.
Thanks for the suggestion, design proposal and for the tech solution.
The Product Team will take a look. Please stay tuned !

Ping @PrestaShop/product-team , I think it's useful to add this feature but it's better to add it in Major release as it contains database changement, do you agree with me?

**Comment by MatShir** [comment]
We can add the feature to a minor or a major, it doesn't matter @hibatallahAouadni ;)
It is a nice feature! The order page has many components, so check with @prestascott before if you want to implement it ;)

Maybe a future improvement, we could add a way to share a file between the merchant and client 🤔

**Comment by TristanLDD** [comment]
Hello @Hlavtox, it's a very interesting feature indeed, the design proposal looks good to me as it's using the UI kit components and the placement is consistant with the rest of the page

**Comment by cheetoozz** [comment]
Dear All  @Hlavtox @MatShir  @TristanLDD @hibatallahAouadni @shakaran 

has this feature been added ? please advise...thank you

regards

**Comment by MatShir** [comment]
Sadly, it is not part of the new release :/
